kratos/input_output/logger_output.cpp

```diff
@@ -37,12 +37,15 @@ namespace Kratos
         mrStream << "_|\\_\\_|  \\__,_|\\__|\\___/ ____/" << std::endl;
         mrStream << "           Multi-Physics "<< KRATOS_VERSION << std::endl;
     }
-    
+
     void LoggerOutput::WriteMessage(LoggerMessage const& TheMessage)
     {
 		auto message_severity = TheMessage.GetSeverity();
-        if (message_severity <= mSeverity)
+        if (TheMessage.WriteInThisRank() && message_severity <= mSeverity)
         {
+			if(TheMessage.IsDistributed())
+				mrStream << "Rank " << TheMessage.GetSourceRank() << ": ";
```

Comment:
After thinking abt this some more I really like that it prints the rank only if running MPI :+1: (i.e. the serial behavior is not affected by this PR)